modules/asg/main.tf

```diff
@@ -0,0 +1,81 @@
+data "aws_ami" "amazon-linux-2" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm-*-x86_64-ebs"]
+  }
+}
+
+resource "aws_launch_template" "t3micro-template" {
+  name_prefix   = "t3micro"
+  image_id      = "ami-0359b3157f016ae46"
+  instance_type = "t3.micro"
+  key_name                    = "MyKeyPair"
+  network_interfaces {
+    associate_public_ip_address = true
+  }
+
+
+  tags = {
+    Name = "1stasg"
+  }
+}
+
+
+resource "aws_autoscaling_group" "MyASG" {
+  
+  desired_capacity   = 3
+  max_size           = 5
+  min_size           = 3
+  vpc_zone_identifier = [data.terraform_remote_state.level1.outputs.public_subnet_id[1], data.terraform_remote_state.level1.outputs.public_subnet_id[0]]
```

Comment:
Instead of terraform_remote_state, please pass subnet_ids as variables

Reply:
I need help with this one